scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1062,6 +1086,23 @@
                               :first-ten-jobs (print-str (->> jobs (take 10) (map :job/uuid) (map str)))}))
       (dissoc compute-cluster->jobs :no-acceptable-compute-cluster))))
 
+(defn distribute-and-handle-jobs
+  "Distributes jobs across compute clusters and applies a given handler for each distribution."
+  [jobs pool-name compute-clusters job->acceptable-compute-clusters-fn handle-jobs-for-cluster-fn]
+  (try
+    (when (seq jobs)
+      (log-structured/info "Preparing for job distribution" {:pool pool-name})
+      (let [compute-cluster->jobs (distribute-jobs-to-compute-clusters jobs pool-name compute-clusters
+                                                                       job->acceptable-compute-clusters-fn)]
+        (log-structured/info "Starting job distribution" {:pool pool-name})
+        (->> compute-cluster->jobs
+             (map handle-jobs-for-cluster-fn)
```

Comment:
The `future` call should happen here, because this code is what derefs the future. (I.e., right now, this code is broken *if* `launch-distributed-job-fn` doesn't return a future.)

Reply:
How does this look?
```
(map (fn [launch-args] (future (launch-distributed-job-fn launch-args))))
```

Reply:
Updated, lmk